marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/ingestion/bulkInputCallerImpl.api

```diff
@@ -0,0 +1,28 @@
+{
+  "endpoint": "/bulkInputCallerImpl.sjs",
+  "params": [ {
+    "name":     "endpointState",
+    "datatype": "jsonDocument",
+    "multiple": false,
+    "nullable": true
+    },{
+      "name":     "workUnit",
+      "datatype": "jsonDocument",
+      "multiple": false,
+      "nullable": true
+      },
+    {
+    "name":     "input",
+    "datatype": "jsonDocument",
+    "multiple": true,
+    "nullable": true
+  } ],
+  "return": {
+    "datatype": "jsonDocument",
+    "multiple": false,
+    "nullable": true
+  },
+  "$bulk":{
```

Comment:
What effect does "$bulk" have?

Reply:
The parameter "inputBatchSize" inside "$bulk" is used by the processInput inside InputEndpointImpl as the number of records(batchSize) it should pick up at a time, process and add to the queue. When the queue size reaches the writeBatchSize, the records get written to the database.